autogen.sh

```diff
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/env bash
```

Comment:
For consistency, since most of the scripts in the project use bash.  
I think it would be convenient to use the same shell everywhere.

It's not particularly important in this file, but in `utils/common.sh` it is important to either use `bash` or remove the `function` keyword.

If you are against it, I can replace `#!/usr/bin/env bash` with `#!/usr/bin/env sh` in the files where it is not crucial.

Reply:
IMHO it would be great to run `autogen.sh` with mostly every possible shell symlinked to `/bin/sh`.

Reply:
I agree. I prefer to keep generic sh for autogen and packages scripts: I don't want to add bash dependency, just to be extra cautious

Reply:
OK. Then for `autogen.sh` and package scripts I'll change the shell back to `sh`.